### PR TITLE
Cog 656 deployment state

### PR DESCRIPTION
**Pull request by dexters1**
## Release Notes

- **New Features**
	- Introduced a new search type, `COMPLETION`, enhancing search functionality.
	- Added a new asynchronous function `query_completion` for processing queries.
	- Implemented a new function `resolve_data_directories` to manage data paths.

- **Enhancements**
	- Improved data retrieval and error handling in dataset management API.
	- Enhanced permission management with user-specific checks and error reporting.
	- Added a new exception class `UnauthorizedDataAccessError` for better security.

- **Documentation**
	- Updated prompts to emphasize brevity in responses.

- **Chores**
	- Expanded module imports to include new classes and functions for better accessibility.

### CODE REVIEW

File: cognee/api/v1/search/search_v2.py [hunk 1]
```diff
@@ -14,11 +14,13 @@
 from cognee.tasks.chunks import query_chunks
 from cognee.tasks.graph import query_graph_connections
 from cognee.tasks.summarization import query_summaries
+from cognee.tasks.compute import query_compute
 
 class SearchType(Enum):
     SUMMARIES = "SUMMARIES"
     INSIGHTS = "INSIGHTS"
     CHUNKS = "CHUNKS"
+    COMPUTE = "COMPUTE"
```
Comment (borisarzentar) [inline cognee/api/v1/search/search_v2.py:23]: I meant `completion` like it usually exists in LLM APIs -> https://llm-engine.scale.com/guides/completions/

File: cognee/modules/data/methods/get_data.py [hunk 2]
```diff
@@ -17,4 +18,7 @@ async def get_data(data_id: UUID) -> Optional[Data]:
     async with db_engine.get_async_session() as session:
         data = await session.get(Data, data_id)
 
+        if data and data.owner_id != user_id:
+            return None
+
```
[the remark this answers is not in the document]
Comment (borisarzentar) [reply cognee/modules/data/methods/get_data.py:23]: @dexters1 This si a good suggestion from coderabbit

File: cognee/api/v1/search/search_v2.py (hunk 1 above)
Comment (dexters1) [reply cognee/api/v1/search/search_v2.py:23]: Will make the change

File: cognee/modules/users/models/GroupPermission.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+from datetime import datetime, timezone
+from sqlalchemy import Column, ForeignKey, DateTime, UUID
+from cognee.infrastructure.databases.relational import Base
+
+class GroupPermission(Base):
+    __tablename__ = "group_permissions"
+
+    created_at = Column(DateTime(timezone = True), default = lambda: datetime.now(timezone.utc))
```
[the remark this answers is not in the document]
Comment (dexters1) [reply cognee/modules/users/models/GroupPermission.py:8]: This also seems to be true, will make a separate task to handle this everywhere

**Review by borisarzentar — APPROVED** [review]

**Pull request merged**